core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -1699,27 +1708,24 @@ public final void addUser(String userName, Object credential, String[] roleList,
                         + "' already exists in the system. Please pick another username.");
             }
 
-
-            List<String> internalRoles = new ArrayList<String>();
-            List<String> externalRoles = new ArrayList<String>();
+            List<String> internalRoles = new ArrayList<>();
+            List<String> externalRoles = new ArrayList<>();
             int index;
-            if (roleList != null) {
-                for (String role : roleList) {
-                    if (role != null && role.trim().length() > 0) {
-                        index = role.indexOf(CarbonConstants.DOMAIN_SEPARATOR);
-                        if (index > 0) {
-                            String domain = role.substring(0, index);
-                            if (UserCoreConstants.INTERNAL_DOMAIN.equalsIgnoreCase(domain)) {
-                                internalRoles.add(UserCoreUtil.removeDomainFromName(role));
-                                continue;
-                            } else if (APPLICATION_DOMAIN.equalsIgnoreCase(domain) ||
-                                    WORKFLOW_DOMAIN.equalsIgnoreCase(domain)) {
-                                internalRoles.add(role);
-                                continue;
-                            }
+            for (String role : roleList) {
```

Comment:
Can the roleList variable be null? If so we would need a null check wrapping the for loop

Reply:
No this cannot be null. There is a null check in a previous line and if null empty array will be assigned.